examples/src/main/kotlin/ai/koog/agents/example/parallelexecution/BestJokeAgent.kt

```diff
@@ -0,0 +1,158 @@
+import ai.koog.prompt.executor.clients.anthropic.AnthropicModels
+import ai.koog.prompt.executor.clients.openai.OpenAILLMClient
+import ai.koog.prompt.executor.clients.openai.OpenAIModels
+import ai.koog.prompt.executor.llms.MultiLLMPromptExecutor
+import ai.koog.prompt.llm.LLMProvider
+import ai.koog.prompt.structure.json.JsonStructuredData
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.Serializable
+
+
+@Serializable
+@LLMDescription("The result of the best joke selection")
+data class JokeWinner(
+    @LLMDescription("Index of the winning joke from 0 to 2")
+    val index: Int,
+    @LLMDescription("The winning joke text")
+    val jokeText: String
+)
+
+
+fun main(args: Array<String>) = runBlocking {
+
+    val jokeSystemPrompt =
+        "You are a comedian. Generate a funny joke about the given topic. Be creative and make it hilarious."
+    val jokeCritiqueSystemPrompt = "You are a comedy critic. Give a critique for the given joke."
+
+    val strategy = strategy("best-joke") {
+        val nodeOpenAI by node<String, String> { topic ->
+            llm.writeSession {
+                model = OpenAIModels.Chat.GPT4o
+                updatePrompt {
+                    system(jokeSystemPrompt)
+                    user("Tell me a joke about $topic.")
+                }
+                val response = requestLLMWithoutTools()
+                response.content
+            }
+        }
+
+        val nodeAnthropicSonnet by node<String, String> { topic ->
+            llm.writeSession {
+                model = AnthropicModels.Sonnet_3_5
+                updatePrompt {
+                    system(jokeSystemPrompt)
+                    user("Tell me a joke about $topic.")
+                }
+                val response = requestLLMWithoutTools()
+                response.content
+            }
+        }
+
+        val nodeAnthropicOpus by node<String, String> { topic ->
+            llm.writeSession {
+                model = AnthropicModels.Opus_3
+                updatePrompt {
+                    system(jokeSystemPrompt)
+                    user("Tell me a joke about $topic.")
+                }
+                val response = requestLLMWithoutTools()
+                response.content
+            }
+        }
+
+        // Define a node to select the best joke
+        val nodeGenerateJokes by parallel(
+            nodeOpenAI, nodeAnthropicSonnet, nodeAnthropicOpus,
+        )
+
+        val nodeTransformJoke by transform<String, String, String> { joke ->
+            "My favorite joke: $joke"
+        }
+
+        val nodeSelectBestJoke by merge<String, String> {
+            selectByIndex { jokes ->
+                // Another LLM (ex: GPT4o) would find the funniest joke:
+                llm.writeSession {
+                    model = OpenAIModels.Chat.GPT4o
+                    updatePrompt {
+                        prompt("best-joke-selector") {
```

Comment:
This part wouldn't work. `updatePrompt` accepts `PromptBuilder` and in it's lambda returns `Unit`. It is meant to add more messages to existing prompt. In this case a new prompt is created and returned, but since the lambda is `Unit`, this operation essentially has no effect and the prompt remains the same.